File: vector-arithmetic/Fortran/src/saxpy.f
        SUBROUTINE saxpy (a, x, dx, y, dy, n)
                ! Scalar a times vector x plus vector y
                ! GVL Algorithm 1.1.2

                ! Input:        a, scalar
                !               x, y, vectors of length n
                !               dx, dy steps between entries of each vector
                ! Output:       ax + y, stored in vector y

                IMPLICIT none
                DOUBLE PRECISION a
                DOUBLE PRECISION x(*), y(*)
                INTEGER dx, dy, n

                INTEGER i, xoff, yoff

                xoff = 1
                yoff = 1
                DO i = 1, n
                        y(yoff) = a * x(xoff) + y(yoff)
                        xoff = xoff + dx
                        yoff = yoff + dy
                END DO
                RETURN
        END SUBROUTINE

File: vector-arithmetic/Matlab/src/saxpy.m
function z = saxpy( a, x, y )
% Scalar a times vector x plus vector y
% GVL Algorithm 1.1.2

% Input:    a, a real scalar
%           x,y real vectors of the same length
% Output:   z = ax + y, a vector of the same size as x and y

n = size(x);
n = n(1);

z = zeros(n,1);
for i = 1:n
    z(i) = a * x(i) + y(i);
end
end

